### ISSUE TITLE
Default test reporter in `test_apps(reporter=)` should behave differently when in testing

### ISSUE THREAD

**Issue by schloerke** [rstudio/shinytest2#189]
By default, use `NULL` to pass through to the default test reporter.

But when `testthat::is_testing()` is `TRUE`, if `reporter` is missing, set to `testthat::FailReporter` to minimize confusion on the screen.

cc @nstrayer

**Comment by schloerke** [comment]
Related code from working with @gadenbuie for `learnr::test_tutorial()`

```r
test_tutorial <- function(
  path,
  test = c("all", "solutions", "tests"),
  ...,
  quiet = TRUE,
  safely = TRUE,
  reporter = NULL
) {
  test <- match.arg(test)

  test_env_vars <- safe_env()
  if (identical(test, "all")) {
    test <- c("solutions", "tests")
  }
  if ("tests" %in% test) {
    test_env_vars <- c(test_env_vars, LEARNR_TEST = "true")
  }
  if ("solutions" %in% test) {
    test_env_vars <- c(test_env_vars, LEARNR_TEST_SOLUTIONS = "true")
  }

  render_call <- rlang::call2(
    .fn = "render",
    input = path,
    output_file = quote(tmpfile),
    quiet = quiet,
    ...,
    .ns = "rmarkdown"
  )

  reporter <- reporter %||%
    testthat::get_reporter() %||%
    if (interactive() && !safely) "progress" else "summary"

  res <-
    if (isTRUE(safely)) {
      reporter_safe <- safe({
        testthat::with_reporter(reporter = "silent", {
          tmpfile <- withr::local_tempfile()
          !!render_call
          testthat::get_reporter()
        })
      }, env = test_env_vars, spinner = FALSE)

      testthat::with_reporter(reporter = reporter, {
        local_reporter <- testthat::get_reporter()
        results <- reporter_safe$expectations()
        for (idx in seq_along(results)) {
          result <- results[[idx]]
          local_reporter$add_result(reporter_safe$.context, test = result$test, result = result)
        }
      })
    } else {
      testthat::with_reporter(
        reporter = reporter,
        withr::with_envvar(test_env_vars, {
          withr::with_tempfile("tmpfile", {
            eval(render_call)
          })
        })
      )
    }

  invisible(res)
}
```


------------------

tl/dr; The inner reporter should use the outer reporter when `safely = FALSE`. This is our situation in shinytest2. 

The key situation above was to use `silent` reporter to gather all expectations. Our situation is similar to `safely = FALSE` above.

So maybe we perform something like...

```r
# testthat::get_reporter() will return `NULL` if no reporter is set; Don't overrride reporter if we can help it
reporter <- reporter %||% testthat::get_reporter() %||% NULL
# Replay all expectations to the current reporter
testthat::with_reporter(
  reporter = reporter,
  # Test the testthat dir
  testthat::test_dir("testthat", ...)
)
```

**Comment by schloerke** [comment]
Fixed in #192
